src/measure_theory/measure/measure_space.lean

```diff
@@ -462,6 +462,29 @@ lemma measure_union_add_inter' (hs : measurable_set s) (t : set α) :
 by rw [union_comm, inter_comm, measure_union_add_inter t hs, add_comm]
 namespace measure
 
+/-- If `u` is a superset of `t` with the same measure (both sets possibly non-measurable), then
+for any measurable set `s` one also has `μ (t ∩ s) = μ (u ∩ s)`. -/
+lemma measure_inter_eq_of_measure_eq {s t u : set α} (hs : measurable_set s)
+  (h : μ t = μ u) (htu : t ⊆ u) (ht_ne_top : μ t ≠ ∞) :
+  μ (t ∩ s) = μ (u ∩ s) :=
+begin
+  rw h at ht_ne_top,
```

Comment:
You could also directly use the hypothesis `hu_ne_top : μ u ≠ ∞`, with `u` instead of `t`

Reply:
But it would require a slightly longer proof for the next lemma, so it would also make sense to keep it as it is.

Reply:
In the applications the assumption is most often on `t`, that's why I formulated it this way.